### ISSUE TITLE
nghttpx: Support TLS-PSK

### ISSUE THREAD

**Issue by tatsuhiro-t** [nghttp2/nghttp2#612]
TLS-PSK has a good use case in a closed private network.

Some items to be considered:
- TLS-PSK ciphersuites supported by OpenSSL is clearly outdated

```
PSK-AES256-CBC-SHA      SSLv3 Kx=PSK      Au=PSK  Enc=AES(256)  Mac=SHA1
PSK-AES128-CBC-SHA      SSLv3 Kx=PSK      Au=PSK  Enc=AES(128)  Mac=SHA1
PSK-RC4-SHA             SSLv3 Kx=PSK      Au=PSK  Enc=RC4(128)  Mac=SHA1
PSK-3DES-EDE-CBC-SHA    SSLv3 Kx=PSK      Au=PSK  Enc=3DES(168) Mac=SHA1
```
- Do we have to rotate pre-shared key just like TLS ticket key rotation?
- bloated settings: current nghttpx settings are for public key based one for both server and client.  To add TLS-PSK, we have to add another set of settings for both server and client.

**Comment by tatsuhiro-t** [comment]
With OpenSSL v1.1.0 (still in pre-release period), we get a bunch of additional TLS-PSK cipher suites.

**Comment by tatsuhiro-t** [comment]
The sad reality is that it seems that no web server supports TLS-PSK at the moment, and node and golang have no built-in support for it.  No application support may cast a shadow for the adoption of PSK even in the its ideal environment like data center.

**Comment by tatsuhiro-t** [comment]
PSK was implemented by #774
